contracts/bonding/BondingManager.sol

```diff
@@ -184,57 +204,42 @@ contract BondingManager is IBondingManager, Manager {
      * @param _to The address of the transcoder to stake towards.
      */
-    function bond(uint256 _amount, address _to) external whenSystemNotPaused returns (bool) {
-        // Current round must be initialized
-        require(roundsManager().currentRoundInitialized());
-        // Must bond to a valid transcoder
-        require(transcoderStatus(_to) == TranscoderStatus.Registered);
-
-        uint256 stakeForTranscoder = _amount;
+    function bond(
+        uint256 _amount,
+        address _to
+    )
+        external
+        whenSystemNotPaused
+        currentRoundInitialized
+        updateDelegatorStake
+        returns (bool)
+    {
+        Delegator storage del = delegators[msg.sender];
+        del.delegationTarget = _to;
+        del.bondedAmount = del.bondedAmount.add(_amount);
+
+        if (delegatorStatus(msg.sender) == DelegatorStatus.Unbonded
+            || (del.delegationTarget != address(0) && _to != del.delegationTarget))
+        {
+            // New delegation target
+            // Set start round
+            del.startRound = roundsManager().currentRound().add(1);
+            // Set delegate block
+            del.delegateBlock = block.number;
+        }
 
-        if (_to == msg.sender) {
-            // Sender is a transcoder bonding to self
-            transcoders[msg.sender].bondedAmount = transcoders[msg.sender].bondedAmount.add(_amount);
-        } else {
-            // Sender is not a transcoder
-            // Update/create delegator
-            Delegator storage del = delegators[msg.sender];
-
-            // Update delegator stake if necessary
-            updateDelegatorStake(msg.sender);
-
-            if (delegatorStatus(msg.sender) == DelegatorStatus.NotRegistered
-                || delegatorStatus(msg.sender) == DelegatorStatus.Unbonded)
-            {
-                // Registering as delegator or bonding to transcoder from the unbonded state
-                // Set start round and delegate block
-                del.startRound = roundsManager().currentRound().add(1);
-                del.delegateBlock = block.number;
-            }
+        if (transcoderStatus(_to) == TranscoderStatus.Registered) {
+            // Bonding to registered transcoder
+            uint256 stakeForTranscoder = _amount;
 
-            if (del.transcoderAddress != address(0) && _to != del.transcoderAddress) {
-                // Delegator is moving bond
-                // Set round start if delegator moves bond to another active transcoder
-                del.startRound = roundsManager().currentRound().add(1);
-                // Decrease former transcoder cumulative stake
-                transcoderPools.decreaseTranscoderStake(del.transcoderAddress, del.bondedAmount);
-                // Stake amount includes delegator's total bonded amount since delegator is moving its bond
+            if (del.delegationTarget != address(0) && _to != del.delegationTarget) {
```

Comment:
Not exactly related to what this PR is about - 

I get what we are trying to do here.  But what are we trying to accomplish by allowing people to "change bond"?  In reality, can't I unbond and then bond to someone else?

Reply:
I think the idea here is to allow users to delegate their bonded tokens to different delegatees without forcing them to first unbond since they will have to wait through the unbonding period before they can re-bond their tokens.